controllers/healthcheck_controller.go

```diff
@@ -321,9 +459,25 @@ func (r *HealthCheckReconciler) watchWorkflowReschedule(ctx context.Context, log
 				hc.Status.FailedCount++
 				hc.Status.Total = hc.Status.SuccessCount + hc.Status.FailedCount
 				hc.Status.LastFailedWorkflow = wfName
-				metrics.MonitorError.With(prometheus.Labels{"healthcheck_name": hc.GetName()}).Inc()
-				metrics.MonitorStartedTime.With(prometheus.Labels{"healthcheck_name": hc.GetName()}).Set(float64(then.Unix()))
-				metrics.MonitorFinishedTime.With(prometheus.Labels{"healthcheck_name": hc.GetName()}).Set(float64(now.Time.Unix()))
+				metrics.MonitorError.With(prometheus.Labels{"healthcheck_name": hc.GetName(), "workflow": "healthcheck"}).Inc()
+				metrics.MonitorStartedTime.With(prometheus.Labels{"healthcheck_name": hc.GetName(), "workflow": "healthcheck"}).Set(float64(then.Unix()))
+				metrics.MonitorFinishedTime.With(prometheus.Labels{"healthcheck_name": hc.GetName(), "workflow": "healthcheck"}).Set(float64(now.Time.Unix()))
+				log.Info("Creating Workflow", "namespace", wfNamespace, "generateNamePrefix", hc.Spec.RemedyWorkflow.GenerateName)
+				err := r.createRBACForWorkflow(log, hc, "remedy")
+				if err != nil {
+					log.Error(err, "Error creating RBAC Permissions for Remedy Workflow")
+					break
+				}
+				generatedWfName, err := r.createSubmitRemedyWorkflow(ctx, log, hc)
+				if err != nil {
+					log.Error(err, "Error creating or submitting remedyworkflow")
+				}
+				r.watchRemedyWorkflow(log, wfNamespace, generatedWfName, hc)
+				//err = r.DeleteServiceAccount(r.kubeclient, hc.Spec.RemedyWorkflow.Resource.ServiceAccount, hc.Spec.RemedyWorkflow.Resource.Namespace)
```

Comment:
Does this need to be commented out? If so, please remove it.

Reply:
yes.. here i am deleting not only service account but also role and role binding.. so for now I have commented it out will clean it up..